lib/s3-compat.js

```diff
@@ -0,0 +1,2229 @@
+ *      S3 request object, the object metadata is directly proportional to the
+ *      HTTP verb and URI that determinate the S3 request operation.
+ *
+ *
+ *  Examples of possible HTTP request and the associated S3 request operation:
+ *
+ *  +----+----------------+----------------+-----------+----------------------+
+ *  |VERB| Path           | S3 Operation   | AddrStyle | S3 Request obj       |
+ *  +----+----------------+----------------+-----------+----------------------+
+ *  |PUT | /mybucket\     |CreateBucketObj | path      | { bucket:"mybucket", |
+ *  |    | /file.txt      |                |           |   object:"file.txt", |
+ *  |    |                |                |           |   operation:"Create",|
+ *  |    |                |                |           |   isS3Request:true,  |
+ *  |    |                |                |           |   mantaPath:null,    |
+ *  |    |                |                |           |   addrStyle:"path" } |
+ *  +----+----------------+----------------+-----------+----------------------+
+ *  |DEL | /mybucket\     |DeleteBucketObj | path      | { bucket:"mybucket", |
+ *  |    | /file.txt      |                |           |   object:"file.txt", |
+ *  |    |                |                |           |   operation:"Delete",|
+ *  |    |                |                |           |   isS3Request:true,  |
+ *  |    |                |                |           |   mantaPath:null,    |
+ *  |    |                |                |           |   addrStyle:"path" } |
+ *  +----+----------------+----------------+-----------+----------------------+
+ *  |PUT | /mybucket\     | UploadPart     | path      | { bucket:"mybucket", |
+ *  |    | /large.zip\    |                |           |   object:"large.zip",|
+ *  |    | ?part=1\       |                |           |   operation:"Upload",|
+ *  |    | &id=abc123     |                |           |   isS3Request:true,  |
+ *  |    |                |                |           |   mantaPath:null,    |
+ *  |    |                |                |           |   addrStyle:"path",  |
+ *  |    |                |                |           |   uploadId:"abc123", |
+ *  |    |                |                |           |   partNumber:1 }     |
+ *  +----+----------------+----------------+-----------+----------------------+
+ *  |GET | /mybucket\     | ListObjsV2     | path      | { bucket:"mybucket", |
+ *  |    | ?list-type=2   |                |           |   object:null,       |
+ *  |    |                |                |           |   operation:"ListV2",|
+ *  |    |                |                |           |   isS3Request:true,  |
+ *  |    |                |                |           |   mantaPath:null,    |
+ *  |    |                |                |           |   addrStyle:"path" } |
+ *  +----+----------------+----------------+-----------+----------------------+
+ *  |GET | bucket.domain\ | GetObj         | vhost     | { bucket:"bucket",   |
```

Comment:
Mention here briefly it's disabled, so when a reader continues they won't whiplash at the first entry in the Notes?